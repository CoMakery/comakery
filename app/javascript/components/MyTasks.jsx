import React from 'react'
import PropTypes from 'prop-types'
import MyTask from './MyTask'
import Icon from './styleguide/Icon'
import styled, {css} from 'styled-components'

const Wrapper = styled.div`
  background-color: white;
`

const Layout = styled.div`
  background-color: white;
  padding: 25px 150px 25px 150px;
  min-height: 50vh;
  max-width: 980px;
  margin: auto;

  @media (max-width: 1024px) {
    padding: 25px 15px 25px 15px;
  }
`

const Header = styled.div`
  height: 140px;
  background-image: url(${require(`src/images/tasks/header_background.jpg`)});
  background-position: center;
  background-size: cover;
  font-family: Montserrat;
  font-size: 30px;
  font-weight: 900;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: #ffffff;
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
  justify-content: center;
`

const PastAwards = styled.div`
  text-align: right;
  width: 100%;

  a {
    font-family: Montserrat;
    font-size: 12px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #0089f4;
    text-decoration: none;
    text-transform: uppercase;

    &:hover {
      text-decoration: underline;
    }
  }
`

const Filter = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 25px 0;
  background-color: #ffffff;

  @media (max-width: 1024px) {
    background: none;
    position: sticky;
    top: 25px;
    overflow-x: auto;
    -webkit-overflow-scrolling: touch;
    margin: 0 -15px;
    padding: 10px;
    z-index: 200;
  }

  &::-webkit-scrollbar {
    display: none;
  }
`

const FilterLink = styled.a`
  text-transform: uppercase;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: #8d9599;
  text-decoration: none;
  line-height: 1.5;
  min-width: 100px;
  padding: 5px;

  &:hover {
    text-decoration: underline;
  }

  ${props => props.current && css`
    color: #0089f4;
    border-bottom: 2px solid #0089f4;
  `}

  @media (max-width: 1024px) {
    background-color: #ffffff;
    box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
    flex: 0 0 auto;
    margin-right: 20px;
  }
`

const SubHeader = styled.div`
  display: flex;
  align-items: center;
  border-bottom: 1px solid #d8d8d8;
  margin-bottom: 35px;
  padding: 10px 0;

  @media (max-width: 1024px) {
    flex-direction: column;
  }
`

const Pagination = styled.div`
  text-align: right;
  width: 100%;
  padding: 15px 0;
`

const ProjectFilter = styled.div`
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  width: 100%;
  text-transform: uppercase;
  color: #8d9599;

  span {
    margin-right: 0.5em;
  }

  img {
    height: 15px;
    margin-left: 0.2em;
    margin-bottom: -3px;
    opacity: 0.2;
  }

  a {
    text-decoration: none;
    color: #0089f4;
    padding: 5px 5px;
    box-shadow: 0 5px 10px 0 rgba(0,0,0,0.1);
    background: white;
    border: solid 1px #d8d8d8;
    display: inline-block;

    &:hover {
      text-decoration: underline;
    }
  }
`

class MyTasks extends React.Component {
  constructor(props) {
    super(props)
    this.state = { ready: false }
  }

  componentDidMount() {
    this.setState({ ready: true })
  }

  render() {
    if (!this.state.ready) {
      return (
        <div className="loading-placeholder" />
      )
    }
    let filter = this.props.filters.find(f => f.current).name
    return (
      <React.Fragment>
        <Wrapper>
          <Header>
            My Tasks
          </Header>

          <Layout>
            <PastAwards>
              <a href={this.props.pastAwardsUrl}>See past awards</a>
            </PastAwards>

            <Filter>
              {this.props.filters.map(filter =>
                <FilterLink key={filter.name} href={filter.url} current={filter.current}>
                  {filter.name}
                  <br />
                  {filter.count}
                </FilterLink>
              )}
            </Filter>

            <SubHeader>
              {this.props.project &&
                <ProjectFilter>
                  <span>Filtered by Project:</span>

                  <a href={location ? location.pathname + '?all=true' : ''}>
                    {this.props.project.title}
                    <Icon name="iconCloseCopy.svg" />
                  </a>
                </ProjectFilter>
              }

              <Pagination dangerouslySetInnerHTML={{__html: this.props.paginationHtml}} />
            </SubHeader>

            {this.props.tasks.map(task =>
              <MyTask
                key={task.id}
                task={task}
                filter={filter}
                displayFilters={!this.props.project && filter === 'ready'}
              />
            )}
          </Layout>
        </Wrapper>
      </React.Fragment>
    )
  }
}

MyTasks.propTypes = {
  tasks         : PropTypes.array.isRequired,
  filters       : PropTypes.array.isRequired,
  project       : PropTypes.object,
  paginationHtml: PropTypes.string.isRequired,
  pastAwardsUrl : PropTypes.string.isRequired
}
MyTasks.defaultProps = {
  tasks         : [],
  filters       : [{'current': true, 'name': '_'}],
  paginationHtml: '',
  pastAwardsUrl : ''
}
export default MyTasks
