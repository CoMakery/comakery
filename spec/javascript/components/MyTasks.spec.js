import React from 'react'
import { mount } from 'enzyme'
import MyTasks from 'components/MyTasks'

describe('MyTasks', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<MyTasks />)

    expect(wrapper).toMatchSnapshot()
  })
})
