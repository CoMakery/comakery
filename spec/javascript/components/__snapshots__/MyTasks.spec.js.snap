// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTasks renders component correctly 1`] = `
<MyTasks
  filters={
    Array [
      Object {
        "current": true,
        "name": "_",
      },
    ]
  }
  paginationHtml=""
  pastAwardsUrl=""
  tasks={Array []}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ckVGcZ",
            "isStatic": true,
            "lastClassName": "diVyLK",
            "rules": Array [
              "
  background-color: white;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ckVGcZ",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-ckVGcZ diVyLK"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-eNQAEJ",
                  "isStatic": true,
                  "lastClassName": "eOHGDW",
                  "rules": Array [
                    "
  height: 140px;
  background-image: url(",
                    "test-file-stub",
                    ");
  background-position: center;
  background-size: cover;
  font-family: Montserrat;
  font-size: 30px;
  font-weight: 900;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: #ffffff;
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
  justify-content: center;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-eNQAEJ",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-eNQAEJ eOHGDW"
            >
              My Tasks
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jKJlTe",
                  "isStatic": true,
                  "lastClassName": "UMbut",
                  "rules": Array [
                    "
  background-color: white;
  padding: 25px 150px 25px 150px;
  min-height: 50vh;
  max-width: 980px;
  margin: auto;

  @media (max-width: 1024px) {
    padding: 25px 15px 25px 15px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jKJlTe",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-jKJlTe UMbut"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-hMqMXs",
                        "isStatic": true,
                        "lastClassName": "klCJRz",
                        "rules": Array [
                          "
  text-align: right;
  width: 100%;

  a {
    font-family: Montserrat;
    font-size: 12px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #0089f4;
    text-decoration: none;
    text-transform: uppercase;

    &:hover {
      text-decoration: underline;
    }
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-hMqMXs",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-hMqMXs klCJRz"
                  >
                    <a
                      href=""
                    >
                      See past awards
                    </a>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kEYyzF",
                        "isStatic": true,
                        "lastClassName": "jSuCvT",
                        "rules": Array [
                          "
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 25px 0;
  background-color: #ffffff;

  @media (max-width: 1024px) {
    background: none;
    position: sticky;
    top: 25px;
    overflow-x: auto;
    -webkit-overflow-scrolling: touch;
    margin: 0 -15px;
    padding: 10px;
    z-index: 200;
  }

  &::-webkit-scrollbar {
    display: none;
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-kEYyzF",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-kEYyzF jSuCvT"
                  >
                    <styled.a
                      current={true}
                      key="_"
                    >
                      <StyledComponent
                        current={true}
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-kkGfuU",
                              "isStatic": false,
                              "lastClassName": "clTSfY",
                              "rules": Array [
                                "
  text-transform: uppercase;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: #8d9599;
  text-decoration: none;
  line-height: 1.5;
  min-width: 100px;
  padding: 5px;

  &:hover {
    text-decoration: underline;
  }

  ",
                                [Function],
                                "

  @media (max-width: 1024px) {
    background-color: #ffffff;
    box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
    flex: 0 0 auto;
    margin-right: 20px;
  }
",
                              ],
                            },
                            "displayName": "styled.a",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-kkGfuU",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <a
                          className="sc-kkGfuU clTSfY"
                        >
                          _
                          <br />
                        </a>
                      </StyledComponent>
                    </styled.a>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-iAyFgw",
                        "isStatic": true,
                        "lastClassName": "UTBMq",
                        "rules": Array [
                          "
  display: flex;
  align-items: center;
  border-bottom: 1px solid #d8d8d8;
  margin-bottom: 35px;
  padding: 10px 0;

  @media (max-width: 1024px) {
    flex-direction: column;
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-iAyFgw",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-iAyFgw UTBMq"
                  >
                    <styled.div
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "",
                        }
                      }
                    >
                      <StyledComponent
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "",
                          }
                        }
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-hSdWYo",
                              "isStatic": true,
                              "lastClassName": "dhqwJl",
                              "rules": Array [
                                "
  text-align: right;
  width: 100%;
  padding: 15px 0;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-hSdWYo",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-hSdWYo dhqwJl"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "",
                            }
                          }
                        />
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MyTasks>
`;
